Can_Balance.rb

```diff
@@ -0,0 +1,38 @@
+=begin  
+Given a non-empty array, return true if there is a place to split the array so that the sum of the numbers on one side is equal 
+to the sum of the numbers on the other side.
+
+canBalance([1, 1, 1, 2, 1]) → true
+canBalance([2, 1, 1, 2, 1]) → false
+canBalance([10, 10]) → true
+=end
+
+def canBalance(array)

```

Comment:
i dont think you got the question fully correct. so you are splitting in the middle and checking if the side on the left is equal to the sum of numbers on the right. your code will fail in this case:

```
print canBalance([4, 1, 1, 2, 1]) 
```

This one should be `true` because you can split the array at position `1`. Then on the left is 4 and the right is 4. 

Another example

```
print canBalance([3, 1, 1, 4]) 
```

Here you can split it at position `2` and then the sum of the left will be 4 and the right will be 4. 

Always think of more examples and test cases than that is in the examples.

Reply:
This seems be a bit tricky problem. I'm not able to build a proper algorithm for it.

Reply:
Algorithm:  
1. Given a list of numbers. 
2. let's take the first and last number from the list. Lets compare the both the numbers if the first number is bigger than the last number. 
3. We should count from the last number towards the first number in the list. 
4. keep counting until we reach the sum equal to first number. 
5. if it gets equal to the first number, then number in between the first number and sum right side of the list would be the split point.

Reply:
So almost there. So the idea would be that we count the numbers from both sides. But when we are counting we have a market that keeps track of where we are on the left side and where we ware on the right side. Then we add numbers and move things one at a time. 

So if the number on the left is 3 and the number on the right most is 1, we keep counting on the right till we get equal to 3 and then count the next number on the left when the right one is equal to 3. So then if we continue like that we will see if we can get them to meet somewhere in the middle and we will be able to split it then.

Reply:
Okay, i think i can do it